tms/models/tms_order.py

```diff
@@ -0,0 +1,336 @@
+    destination_location = fields.Char(string="Destination")
+
+    driver_id = fields.Many2one(
+        "tms.driver",
+        string="Driver",
+        compute="_compute_vehicle_id_set_driver",
+        store=True,
+        readonly=False,
+    )
+    vehicle_id = fields.Many2one("fleet.vehicle", string="Vehicle")
+    tms_team_id = fields.Many2one("tms.team", string="Team")
+    crew_id = fields.Many2one("tms.crew", string="Crew")
+
+    stage_id = fields.Many2one(
+        "tms.stage",
+        string="Stage",
+        index=True,
+        copy=False,
+        domain="[('stage_type', '=', 'order')]",
+        default=lambda self: self._default_stage_id(),
+        group_expand="_read_group_stage_ids",
+        ondelete="set null",
+    )
+
+    scheduled_date_start = fields.Datetime(
+        string="Scheduled Start", default=datetime.now()
+    )
+    scheduled_duration = fields.Float(
+        help="Scheduled duration of the work in" " hours",
+        compute="_compute_scheduled_duration",
+        store=True,
+        readonly=False,
+    )
+    scheduled_date_end = fields.Datetime(
+        string="Scheduled End",
+        compute="_compute_scheduled_date_end",
+        store=True,
+        readonly=False,
+    )
+
+    start_trip = fields.Boolean(readonly=True)
+    end_trip = fields.Boolean(readonly=True)
+    date_start = fields.Datetime()
+    date_end = fields.Datetime(compute="_compute_date_end", store=True, readonly=False)
+    duration = fields.Float(
+        string="Trip Duration", compute="_compute_duration", store=True, readonly=False
+    )
+
+    diff_duration = fields.Float(
+        readonly=True, string="Scheduled Duration - Actual Duration"
+    )
+
+    time_uom = fields.Many2one(
+        "uom.uom",
+        domain="[('category_id', '=', 'Working Time')]",
+        default=lambda self: self._default_time_uom_id(),
+    )
+
+    def _default_time_uom_id(self):
+        # Fetch the value of default_time_uom from settings
+        default_time_uom_id = (
+            self.env["ir.config_parameter"].sudo().get_param("tms.default_time_uom")
+        )
+
+        # Return the actual record based on the ID retrieved from settings
+        if default_time_uom_id:
+            return self.env["uom.uom"].browse(int(default_time_uom_id))
+        else:
+            # If no default_time_uom is set, return None or a default value
+            return self.env.ref("uom.product_uom_hour", raise_if_not_found=False)
+
+    def _default_stage_id(self):
+        stage = self.env["tms.stage"].search(
+            [
+                ("stage_type", "=", "order"),
+            ],
+            order="sequence asc",
+            limit=1,
+        )
+        if stage:
+            return stage
+        raise ValidationError(_("You must create an TMS order stage first."))
+
+    @api.depends("route")
+    def _compute_route_id(self):
+        for order in self:
+            if order.route:
+                order.origin_id = None
```

Comment:
`False` is a better in Odoo's ORM context.

You can update several properties with :

```suggestion
                order.update({
                    "origin_id": False,
                    "destination_id": False
                })
```

Reply:
Moreover, you cannot update fields that are note computed in a compute method. The best is to transform those fields in `compute=True`/`store=True`/`readonly=False`fields.